core-rust/state-manager/src/mempool/pending_transaction_result_cache.rs

```diff
@@ -338,7 +358,7 @@ impl PendingTransactionRecord {
     pub fn should_accept_into_mempool(
         self,
         check: CheckMetadata,
-    ) -> Result<Box<ValidatedNotarizedTransactionV1>, MempoolAddRejection> {
+    ) -> Result<(Box<ValidatedNotarizedTransactionV1>, StateVersion), MempoolAddRejection> {
```

Comment:
```suggestion
    ) -> Result<SomeWellNamedType, MempoolAddRejection> {
```

I only suggest this since `SomeWellNamedType` would be a nice place to explain what is the returned state version.